documentation/src/docs/pravega-concepts.md

```diff
@@ -11,429 +11,312 @@ You may obtain a copy of the License at
-of the shared state to make access to the data really fast for the application.
- Any changes to shared state are written through the StateSynchronizer to the
-Stream keeping track of all changes to the shared state.
- Each application instance uses the State Synchronizer to stay up to date with
-changes by pulling updates to shared state and modifying the local copy of the
-data.  Consistency is maintained through a conditional append style of updates
+map with thousands of different key value pairs. In Pravega, managing the state of ReaderGroups and distribution of Readers throughout the network is implemented using State Synchronizer.
+
+An application developer creates a State Synchronizer on a Stream similar to the creation of Writer. The State Synchronizer keeps a local copy of the shared state and allows faster access to the data in the application. State Synchronizer keeps track of all the changes happening in the shred state and responsible for performing any modification to the shared state in the Stream. Each application instance uses the State Synchronizer, to remain updated with the
+changes by pulling updates to the shared state and modifies the local copy of the
+data. Consistency is maintained through a conditional append style of updates
 to the shared state through the State Synchronizer, making sure that updates are
 made only to the most recent version of the shared state.
 
-The State Synchronizer can occasionally be "compacted", compressing and removing
-old state updates so that only the most recent version of the state is kept in
-the backing stream.  This feature helps app developers make sure that shared
-state does not grow unchecked.
-
- 
+The State Synchronizer can occasionally be "compacted", by compressing and removing
+older updates and retains only the recent version of the state in the backing Stream. This feature assures the application developers, that the shared state does not grow unchecked.
 
-State Synchronizer works best when most updates to shared state are small in
-comparison to the total data size being stored, allowing them to be written as
+State Synchronizer works effectively when most updates to shared state are small in
+comparison to the total data size being stored. This can be achieved by allowing them to be written as
 small deltas. As with any optimistic concurrency system, State Synchronizer is
-not at its best when many processes are all attempting to simultaneously update
-the same piece of data.
+not at its best when many processes attempts for simultaneous updates on the same piece of data.
 
-For more details on working with State Synchronizers, see [Working with Pravega:
+For more details on working with State Synchronizers, please see [Working with Pravega:
 State Synchronizer](state-synchronizer.md).
 
 ## Architecture
 
-The following figure depicts the components deployed by Pravega:
+The following figure depicts the components deployed by Pravega
```

Comment:
The sentence should end wither with `:` or with another punctuation symbol.

Reply:
Accepted Suggestion